docs/input.rst

```diff
@@ -223,10 +199,414 @@ a brace character in the literal text, it can be escaped by doubling:
 ``{{`` and ``}}``. [Python_Format]_
 
 
+Attributes
+----------
+
+Annotations or attributes apply to specific arguments or results.
+They describe semantic behavior for an argument.
+An attribute may be set to true by listing its name or
+it may have a value in parens::
+
+    - decl: Class1()  +name(new)
+    - decl: void Sum(int len, int *values+dimension+intent(in))
+    - decl: const std::string getName() +len(30)
+
+Attributes may also be added external to *decl*::
+
+    - decl: void Sum(int len, int *values)
+      attrs:
+          values:
+              dimension: True
+              intent: in  
+    - decl: const std::string getName()
+      fattrs:
+          len: 30
+  
+
+allocatable
+^^^^^^^^^^^
+
+Sometimes it is more convient to have the wrapper allocate an
+``intent(out)`` array before passing it to the C++ function.  This can
+be accomplished by adding the *allocatable* attribute.  For example the
+C++ function ``cos_doubles`` takes the cosine of an ``intent(in)``
+argument and assigns it to an ``intent(out)`` argument::
+
+    void cos_doubles(double *in, double *out, int size)
+    {
+        for(int i = 0; i < size; i++) {
+            out[i] = in[i] * 2.;
```

Comment:
for consistency with the function name, this should probably be ``out[i] = cos(in[i])``

(Great example though!)

Reply:
This example was lifted from a web page which compared various ways to wrap.  I changed it to be a multiplication to minimize fuzz issues when comparing real numbers.  I update the test to actually use cos and fixed the comparison issues in the tests.